### PR TITLE
[FIXED JENKINS-23609] "Successful" and "Failed" links do not work

### PR DESCRIPTION
**Pull request by stephenc**
Seemed to not actually be providing the run result data to the matrix. Unclear how this could ever have worked

@reviewbybees

### CODE REVIEW

**Comment by rsandell** [comment]
:+1:

**Comment by stephenc** [comment]
@ikedam this pull request has now successfully completed the CloudBees internal processes and we are now asking you, as the maintainer of the plugin for your feedback with a view towards merging and cutting a new release with this fix

File: src/main/resources/hudson/plugins/matrix_configuration_parameter/MatrixCombinationsParameterDefinition/index.groovy [hunk 1]
```diff
@@ -1,13 +1,10 @@
 package hudson.plugins.matrix_configuration_parameter.matrixcombinationparameterDefinition
 
-import hudson.matrix.AxisList
-import hudson.matrix.Combination
-import hudson.matrix.Layouter
-import hudson.matrix.MatrixBuild
-import hudson.matrix.MatrixProject
+import hudson.matrix.*
+import hudson.model.Result
+import hudson.plugins.matrix_configuration_parameter.MatrixCombinationsParameterDefinition
```
Comment (ikedam) [inline src/main/resources/hudson/plugins/matrix_configuration_parameter/MatrixCombinationsParameterDefinition/index.groovy]: You may minimize changes in pull requests not to conflict with other pull requests.
(I don't think you need to update and fix this pull request as there's no other pull requests)

**Comment by ikedam** [comment]
This feature have never worked. It was broken already in the first release
(omer727/Matrix-Combinations-Jenkins-Plugin#4).

I'm not sure `document.parameters` works in any web browsers.
Do you know about that?

**Comment by ikedam** [comment]
I want also [JENKINS-27233](https://issues.jenkins-ci.org/browse/JENKINS-27233) fixed in the next release, but I don't have time to work for that soon.

Would you (or CloudBees) work for that?
If not, I'll make a new release only with this fix.

File: src/main/resources/hudson/plugins/matrix_configuration_parameter/MatrixCombinationsParameterDefinition/index.groovy [hunk 2]
```diff
@@ -66,7 +63,7 @@ private void drawMainLinksJS(String nameIt) {
             "var element = document.parameters.elements[i];\n" +
```
Comment (stephenc) [inline src/main/resources/hudson/plugins/matrix_configuration_parameter/MatrixCombinationsParameterDefinition/index.groovy]: @ikedam notice the use of `document.{formname}.elements` here which is all that I copy down below

**Comment by stephenc** [comment]
@ikedam so `document.parameters` is just the accessing of the form named `parameters` that is part of the "contract" of parameters.

I had a look at [JENKINS-27233](https://issues.jenkins-ci.org/browse/JENKINS-27233) and I just cannot get my head around how to avoid the looking up matrix project. Maybe I'm just not seeing the whole context.

Anyway, it would be good if you can merge this. If you think it will be more than two weeks before you can cut a release I would favour releasing this one early rather than wait 2+ weeks for a release with [JENKINS-27233](https://issues.jenkins-ci.org/browse/JENKINS-27233) as an addition

**Comment by ikedam** [comment]
> `document.parameters` is just the accessing of the form named `parameters`

I got it! Thanks!

OK, I'll merge this change and make a new release soon (without [JENKINS-27233](https://issues.jenkins-ci.org/browse/JENKINS-27233)).

**Pull request merged**